source/ColumnChooserPanel.cpp

```diff
@@ -0,0 +1,129 @@
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ColumnChooserPanel.h"
+
+#include "FillShader.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "GameData.h"
+#include "Information.h"
+#include "Interface.h"
+#include "Preferences.h"
+#include "Sprite.h"
+#include "SpriteSet.h"
+#include "SpriteShader.h"
+#include "UI.h"
+#include "text/WrappedText.h"
+
+#include <numeric>
+
+using namespace std;
+
+ColumnChooserPanel::ColumnChooserPanel(const vector<PlayerInfoPanel::SortableColumn> &columns,
+	InfoPanelState *panelState)
+	: columns(columns), panelState(panelState)
+{
+	SetInterruptible(false);
+}
+
+
+
+void ColumnChooserPanel::Draw()
+{
+	if(GetUI()->IsTop(this))
+		DrawBackdrop();
+
+	zones.clear();
+
+	Information info;
+	const Interface *columnChooser = GameData::Interfaces().Get("columns menu");
+	info.SetCondition("columns menu open");
+	columnChooser->Draw(info, this);
+
+	const Font &font = FontSet::Get(14);
+	const Color * const dim = GameData::Colors().Get("dim");
+	const Color * const medium = GameData::Colors().Get("medium");
+	const Color * const bright = GameData::Colors().Get("bright");
+	const Sprite *box[] = {SpriteSet::Get("ui/unchecked"), SpriteSet::Get("ui/checked")};
+
+	Point topLeft(270., -280.);
+	const Point boxSize = Point(box[0]->Width(), box[0]->Height());
+	static const Point ROW_ADVANCE(0., 20.);
+	const set<const string> visibleColumns = panelState->VisibleColumns();
+	auto isVisible = [&](string name){ return visibleColumns.find(name) != visibleColumns.end(); };
+	const float availableWidth = 727 - accumulate(columns.begin(), columns.end(), 0,
+		[&](int acc, PlayerInfoPanel::SortableColumn column) {
+			return acc + (isVisible(column.name) ? column.layout.width : 0);
+		});
+	for(PlayerInfoPanel::SortableColumn &column : columns)
+	{
+		Rectangle zoneBounds = Rectangle::FromCorner(topLeft, Point(220., ROW_ADVANCE.Y()));
+
+		bool visible = isVisible(column.name);
+		bool enabled = visible || column.layout.width <= availableWidth;
+		bool hover = zoneBounds.Contains(hoverPoint);
+
+		Rectangle spriteBounds = Rectangle::FromCorner(topLeft, boxSize);
+		SpriteShader::Draw(box[visible], spriteBounds.Center());
+
+		Point textPos = topLeft + Point(box[0]->Width(), 2.);
+		font.Draw(column.checkboxLabel, textPos, enabled ? hover ? *bright : *medium : *dim);
+
+		if(enabled)
+			zones.emplace_back(ClickZone(zoneBounds, column.name));
+
+		topLeft = topLeft + ROW_ADVANCE;
```

Comment:
there is no `operator+=` for Point.

Reply:
https://github.com/endless-sky/endless-sky/blob/33bf24da36be9e73573523d1b43fa0bb095d0888/source/Point.cpp#L78-L87

Reply:
Waiiiit wuuut?
I tried it and it didn't work!